src/arrayMethodJoin.js

```diff
@@ -4,8 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    } else if (this.length === 1) {
+      return String(this[0]);
+    }
+
+    let result;
+    const separatorToString = String(separator);
+
+    if (this[0] === undefined || this[0] === null) {
+      result = '';
+    } else {
+      result = this[0];
+    }
```

Comment:
In both cases you are returning a value, ternary exists for this purposes
```suggestion

    const result = (this[0] === undefined || this[0] === null) ? '' : String(this[0]);
```